airflow/providers/microsoft/azure/operators/synapse.py

```diff
@@ -49,6 +49,7 @@ class AzureSynapseRunSparkBatchOperator(BaseOperator):
     template_fields: Sequence[str] = (
         "azure_synapse_conn_id",
         "spark_pool",
+        "payload"
```

Comment:
Just wondering how it supposed to be work? According to the `payload` annotation it is expected to get `azure.synapse.spark.models.SparkBatchJobOptions` object

https://github.com/apache/airflow/blob/9d8c77e447f5515b9a6aa85fa72511a86a128c28/airflow/providers/microsoft/azure/operators/synapse.py#L63

For allow render templates for custom types it is required to have templated_field class method, see: [Jinja Templating](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#jinja-templating)

Reply:
I would essentially look something like this. The templating allows to pass execution and task instance related attributes to the job.
In the following example I pass the logical execution datetime to the job using `{{ ts }}`.
```
    payload_example = {
        "name": "synapse_spark_def",
        "file": "abfss://<container>@<azstorageaccount>.dfs.core.windows.net/entrypoints/<filename>.py",
        "args": [
            "--exec_time",
            "{{ ts }}",    # Trying to pass something like a timestamp to job is where rendering will happen
        ],
        "jars": [],
        "pyFiles": [],
        "files": [],
        "conf": {
            "spark.dynamicAllocation.enabled": "false",
            "spark.dynamicAllocation.minExecutors": "3",
            "spark.dynamicAllocation.maxExecutors": "4",
        },
        "numExecutors": 3,
        "executorCores": 4,
        "executorMemory": "28g",
        "driverCores": 4,
        "driverMemory": "28g"
    }
```